app/models/git/exercise.rb

```diff
@@ -1,64 +1,66 @@
 module Git
   class Exercise
     def self.for_solution(solution)
-      repo = solution.track.repo
-      repo.exercise(
-        solution.exercise.slug,
+      new(
+        solution.track.slug,
+        solution.git_slug,
         solution.git_sha
       )
     end
 
-    def initialize(repo, slug, commit, track_config)
-      @repo = repo
-      @slug = slug
-      @commit = commit
-      @track_config = track_config
+    def initialize(track_slug, exercise_slug, git_sha)
+      @track_slug = track_slug
+      @exercise_slug = exercise_slug
+      @git_sha = git_sha
     end
 
-    def filepaths
-      files.map { |defn| defn[:full] }
+    def data
```

Comment:
You'll likely have already considered this, but is it worth memoizing this? I don't know if there are places where callers will call this method more than once. If not, ignore this.

Reply:
Nice. yes. Done :)